File: Beginner/Python/Array/RotateImage.py
class Solution(object):
    def rotate(self, matrix):
        """
            :type matrix: List[List[int]]
            :rtype: void Do not return anything, modify matrix in-place instead.
            """
        for i in range(len(matrix) / 2):
            for j in range(i, len(matrix) - i - 1):
                temp = matrix[i][j]
                for _ in range(4):
                    matrix[j][-(i + 1)], temp = temp, matrix[j][-(i + 1)]
                    i, j = j, -(i + 1)

File: Beginner/Java/Array/RotateImage.java
class Solution {
    public void rotate(int[][] matrix) {
        int row = matrix.length;
        int col = matrix[0].length;
        for (int i = 0; i < row / 2; i ++) {
            for (int j = i; j < col - i - 1; j ++) {
                int temp = matrix[i][j];
                for (int k = 0; k < 4; k ++) {
                    int t = matrix[j][col - i - 1];
                    matrix[j][col - i - 1] = temp;
                    temp = t;
                    t = i; i = j; j = row - t - 1;
                }
            }
        }
    }
}
